docs/ProfileDeveloperGuide.md

```diff
@@ -0,0 +1,430 @@
+# Table of Contents
+1. [Profiles](#Profiles)
+    1. [Fields](#Fields)
+    2. [Rules](#Rules)
+    3. [Persistence](#Persistence)
+    4. [Creation](#Creation)
+
+2. [Data types](#Data-Types)
+    1. [Numeric](#Numeric)
```

Comment:
Note that there may well be merge conflicts with the changes Matt has been working on

Reply:
What pull request/branch are this changes under?

Reply:
These changes are in master now